docs/api/python/text/text.md

```diff
@@ -0,0 +1,213 @@
+>>> len(token_indexer2)
+2
+>>> token_indexer2.token_to_idx
+{'<unk>': 0, 'c': 1}
+>>> token_indexer2.idx_to_token
+['<unk>', 'c']
+>>> token_indexer2.unknown_token
+'<unk>'
+```
+
+## Text token embedding
+
+The text token embedding builds indices for text tokens. Such indexed tokens can
+be used by instances of `mxnet.text.embedding.TokenEmbedding` and
+`mxnet.text.glossary.Glossary`.
+
+To load token embeddings from an externally hosted pre-trained token embedding
+file, such as those of GloVe and FastText, use
+`TokenEmbedding.create(embedding_name, pretrained_file_name)`. To get all the
+available `embedding_name` and `pretrained_file_name`, use
+`TokenEmbedding.get_embedding_and_pretrained_file_names()`.
+
+Alternatively, to load embedding vectors from a custom pre-trained text token
+embedding file, use `mxnet.text.embeddings.CustomEmbedding`.
+
+
+```eval_rst
+.. currentmodule:: mxnet.text.embedding
+.. autosummary::
+    :nosignatures:
+
+    TokenEmbedding
+    Glove
+    FastText
+    CustomEmbedding
+```
+
+```python
+>>> from mxnet.text.embedding import TokenEmbedding
+>>> TokenEmbedding.get_embedding_and_pretrained_file_names()
+{'glove': ['glove.42B.300d.txt', 'glove.6B.50d.txt', 'glove.6B.100d.txt',
+'glove.6B.200d.txt', 'glove.6B.300d.txt', 'glove.840B.300d.txt',
+'glove.twitter.27B.25d.txt', 'glove.twitter.27B.50d.txt',
+'glove.twitter.27B.100d.txt', 'glove.twitter.27B.200d.txt'],
+'fasttext': ['wiki.en.vec', 'wiki.simple.vec', 'wiki.zh.vec']}
+>>> glove_6b_50d = TokenEmbedding.create('glove',
+...                                      pretrained_file_name='glove.6B.50d.txt')
+>>> len(glove_6b_50d)
+400001
+>>> glove_6b_50d.vec_len
+50
+>>> glove_6b_50d.token_to_idx['hi']
+11084
+>>> glove_6b_50d.idx_to_token[11084]
+'hi'
+>>> # 0 is the index for any unknown token.
+... glove_6b_50d.idx_to_vec[0]
+
+[ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
+  ...
+  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
+<NDArray 50 @cpu(0)>
+>>> glove_6b_50d.get_vecs_by_tokens('<unk$unk@unk>')
+
+[ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
+  ...
+  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
+<NDArray 50 @cpu(0)>
+>>> glove_6b_50d.get_vecs_by_tokens(['<unk$unk@unk>', '<unk$unk@unk>'])
+
+[[ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
+   ...
+   0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
+ [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
+   ...
+   0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]]
+<NDArray 2x50 @cpu(0)>
+
+```
+
+## Implement a new text token embedding
+
+For ``optimizer``, create a subclass of `mxnet.text.embedding.TokenEmbedding`.
+Also add ``@TokenEmbedding.register`` before this class. See
+[`embedding.py`](https://github.com/dmlc/mxnet/blob/master/python/mxnet/text/embedding.py)
+for examples.
+
+
+## Glossary
```

Comment:
Consider moving `glossary` to appear earlier in the doc, since it has the most comprehensive set of features.

Reply:
resolved.